File: CS/CustomFunctionForExpressionEditorExample/Form1.cs
using System;

namespace CustomFunctionForExpressionEditorExample
{
    public partial class Form1 : DevExpress.XtraEditors.XtraForm
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e)
        {
            DevExpress.XtraReports.Expressions.CustomFunctions.Register(new NewLineConstant());
            DevExpress.XtraReports.Expressions.CustomFunctions.Register(new CustomFormatFunction());
            reportDesigner1.AddService(
                typeof(DevExpress.XtraReports.UserDesigner.ReportExpressionEditorCustomizationService),
                new CustomReportExpressionEditorCustomizationService());
            reportDesigner1.OpenReport(new Reports.XtraReport1());
        }
    }
}

File: VB/CustomFunctionForExpressionEditorExample/Form1.vb
Imports System
Imports DevExpress.XtraReports.UserDesigner

Namespace CustomFunctionForExpressionEditorExample
    Partial Public Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub
#Region "#formload"
        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            DevExpress.XtraReports.Expressions.CustomFunctions.Register(New NewLineConstant())
            DevExpress.XtraReports.Expressions.CustomFunctions.Register(New CustomFormatFunction())
            reportDesigner1.AddService(GetType(ReportExpressionEditorCustomizationService),
                                       New CustomReportExpressionEditorCustomizationService())
            reportDesigner1.OpenReport(New XtraReport1())
        End Sub
#End Region
    End Class
End Namespace
